lib/tbot/config/output_database.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gravitational/trace"
+	"golang.org/x/exp/slices"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/tbot/bot"
+	"github.com/gravitational/teleport/lib/tbot/identity"
+)
+
+const DatabaseOutputType = "database"
+
+type DatabaseSubtype string
+
+var (
+	// UnspecifiedDatabaseSubtype is the unset value and the default. This
+	// should work for most databases.
+	UnspecifiedDatabaseSubtype DatabaseSubtype = ""
+	// TLSDatabaseSubtype is for databases that require specifically named
+	// outputs: tls.key, tls.crt and tls.cas
+	TLSDatabaseSubtype DatabaseSubtype = "tls"
```

Comment:
TLS as a subtype seems kinda awkward - I'd be open to some better naming or thoughts here. This could just be broken out into a configuration key on Output.

Reply:
Maybe some food for thought, `tls` (and most of our types here) correspond to `tctl auth sign --format=<type>`. I think we can certainly rethink how this works, but there may also be some value in consistency between our tools.

Reply:
Changed this to `format` instead, as it aligns better with other tools and also conveys what it does more accurately: 788495648b